dm/master/server.go

```diff
@@ -1944,3 +1967,35 @@ func (s *Server) workerArgsExtractor(args ...interface{}) (workerrpc.Client, str
 
 	return cli, worker, nil
 }
+
+// HandleHTTPApis handles http apis and translate to grpc request
+func (s *Server) HandleHTTPApis(ctx context.Context, mux *http.ServeMux) error {
+	// MasterAddr's format may be "host:port" or "":port"
+	_, port, err := net.SplitHostPort(s.cfg.MasterAddr)
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	opts := []grpc.DialOption{grpc.WithInsecure()}
+	conn, err := grpc.DialContext(ctx, "127.0.0.1:"+port, opts...)
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	gwmux := runtime.NewServeMux()
+	err = pb.RegisterMasterHandler(ctx, gwmux, conn)
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	mux.Handle("/apis/", wsproxy.WebsocketProxy(
```

Comment:
some code is same with etcd https://github.com/etcd-io/etcd/blob/89d400211de4f5dc0715fd08564362edc8b18d7a/embed/serve.go#L162